doc/modules/preprocessing.rst

```diff
@@ -381,10 +381,105 @@ The normalizer instance can then be used on sample vectors as any transformer::
+not desired (i.e. the set of browsers was ordered arbitrarily).
+
+One possibility to convert categorical features to features that can be used
+with scikit-learn estimators is to use a one-of-K or one-hot encoding, which is
+implemented in :class:`OneHotEncoder`.  This estimator transforms each
+categorical feature with ``m`` possible values into ``m`` binary features, with
+only one active.
+
+Continuing the example above::
+
+  >>> enc = preprocessing.OneHotEncoder()
+  >>> enc.fit([[0, 0, 3], [1, 1, 0], [0, 2, 1], [1, 0, 2]])  # doctest: +ELLIPSIS
+  OneHotEncoder(categorical_features='all', dtype=<... 'numpy.float64'>,
+         handle_unknown='error', n_values='auto', sparse=True)
+  >>> enc.transform([[0, 1, 3]]).toarray()
+  array([[ 1.,  0.,  0.,  1.,  0.,  0.,  0.,  0.,  1.]])
+
+By default, how many values each feature can take is inferred automatically from the dataset.
+It is possible to specify this explicitly using the parameter ``n_values``.
+There are two genders, three possible continents and four web browsers in our
+dataset.
+Then we fit the estimator, and transform a data point.
+In the result, the first two numbers encode the gender, the next set of three
+numbers the continent and the last four the web browser.
+
+Note that, if there is a possibilty that the training data might have missing categorical
+features, one has to explicitly set ``n_values``. For example,
+
+    >>> enc = preprocessing.OneHotEncoder(n_values=[2, 3, 4])
+    >>> # Note that there are missing categorical values for the 2nd and 3rd
+    >>> # features
+    >>> enc.fit([[1, 2, 3], [0, 2, 0]])  # doctest: +ELLIPSIS
+    OneHotEncoder(categorical_features='all', dtype=<... 'numpy.float64'>,
+           handle_unknown='error', n_values=[2, 3, 4], sparse=True)
+    >>> enc.transform([[1, 0, 0]]).toarray()
+    array([[ 0.,  1.,  1.,  0.,  0.,  1.,  0.,  0.,  0.]])
+
+See :ref:`dict_feature_extraction` for categorical features that are represented
+as a dict, not as integers.
+
+.. _discretization:
+
+Discretization
+==============
+
+`Discretization <https://en.wikipedia.org/wiki/Discretization_of_continuous_features>`_
+(otherwise known as quantization or binning) provides a way to partition continuous
+features into discrete values. Certain datasets with continuous features
+may benefit from discretization, because discretization can transform the dataset
+of continuous attributes to one with only nominal attributes.
+
+K-bins discretization
+---------------------
+
+Scikit-learn allows users to discretize features into ``k``
+equal width bins using a :class:`KBinsDiscretizer`::
+
+  >>> X = np.array([[ -3., 5., 15 ],
+  ...               [  0., 6., 14 ],
+  ...               [  6., 3., 11 ]])
+  >>> est = preprocessing.KBinsDiscretizer(n_bins=[3, 3, 2]).fit(X)
+
+In this example, the "bin edges" (boundaries for the discretization criteria)
+for each of the three features are as follows:
+
+ - ``0`` and ``3``
+ - ``4`` and ``5``
+ - ``13``
 
```

Comment:
Would it make sense to first introduce the `bin_width_` and then the "bin edges" as interval. For instance something like:

```
>>> est.bin_width_
array([ 3.,  1.,  2.])

For each feature, the bin width is computed during `fit` and together with the number
of bins, they will define the bin intervals. Therefore, for the current example, these
intervals are defined as:

- feature #1: :math:`{[-3, 0), [0, 3), [3, 6]}`
- feature #2: :math:`{[3, 4), [4, 5), [5, 6]}`
- feature #3: :math:`{[11, 13), [13, 15]}`

```